data/basics/address.go

```diff
@@ -87,15 +87,15 @@ func UnmarshalChecksumAddress(address string) (Address, error) {
 	return short, nil
 }
 
-// String returns a string representation of ChecksumAddress
-func (addr *ChecksumAddress) String() string {
+// String returns a string representation of checksumAddress
+func (addr *checksumAddress) String() string {
 	var addrWithChecksum []byte
 	addrWithChecksum = append(addr.shortAddress[:], addr.checksum...)
 	return base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(addrWithChecksum)
 }
 
 func (addr Address) String() string {
-	return fmt.Sprintf("%v", crypto.Digest(addr))
+	return addr.getChecksumAddress().String()
```

Comment:
move the code in here and delete `checksumAddress` ?

Reply:
can do, I like the current breakdown since you can unit test checksumAddress on its own inside of the basics package.